### ISSUE TITLE
Encrypted backup file cannot be fully decrypted using offline tool

### ISSUE THREAD

**Issue by Matei59** [standardnotes/forum#1318]
Hello. I am SN user with a 5 year plan, subscription started in Oct 2020. I am using the SN web and iOS clients.
I have automatic daily backup enabled to Dropbox. 

Today I tried to decode a backup file (txt) saved yesterday using the offline script: 
  https://raw.githubusercontent.com/standardnotes/decrypt/master/dist/decrypt.html 
and received an error:
   "25 items could not be decrypted. Please make sure the password you entered is correct, and try again." 

I checked the resulting zip archive [A] and in the "Note" folder there were some notes decoded correctly, but some notes were missing. I logged in the SN Web client and exported an encrypted backup from the web interface [B]. 
I tried to decode this backup (txt) using the offline script and it worked ok -- all the notes were decoded.

I checked also the "Tag" folder in the decoded backup -- in the version [A] produced after the above error, I have 4 tags. In the other one [B], I have 5 tags. The tag file which appears only in the backup [B] has this entry:
  "appData": {
    "org.standardnotes.sn": {
      "client_updated_at": "1970-01-01T00:00:00.000Z"
    }
I don't know if "1970" is related to the issue, but the only pattern I see regarding the missing notes from set [A] is that they are older and were not updated recently. I think the most recent one was updated in Nov 2020.

Also, I don't know if this issue is related to the v4 upgrade described at https://docs.standardnotes.org/specification/encryption/ -- but the "good" encrypted file which generated [B] contains this at the beginning:
   "version": "004",
and the file which produced archive [A] does not have any version info at the start.

In summary, it seems that the backup mechanism used for the automatic export is different than the one used in the web interface and it generates data loss. 

I consider this a severe and major issue, because it renders the backup files partially unusable. 

Also, it creates a false sense of safety for the user -- the backup files exist to protect the user in case anything goes wrong with the SN service (https://standardnotes.org/help/4/what-happens-to-my-data-if-standard-notes-disappears) --  but for this to work, the backups themselves should be reliable. 

I am looking forward to a solution to this problem. 
While I understand that the old backups are corrupted due to the above issue, I would like to know if this can be fixed for future backups. Basically, the cloud backups should be similar to those produced with the web interface.

Thank you.

**Comment by Matei59** [comment]
Hello, I checked the old backup files saved in Dropbox and starting with 2021-01-08, the backup files seem to have the above problem, i.e. they cannot be fully decrypted (25 notes are missing). 

The automatic backup saved on 2021-01-07 is decrypted correctly, without errors -- on Jan 7 there are 31 notes saved, on Jan 8 there are only 14 notes saved in "Note" folder in the decrypted zip archive. 

As an observation, the backup txt from Jan 7 ends with  "version": "003" and the backup from Jan 8 ends with "version": "004".

Looking into the encrypted txt file from Jan 8, it seems the number of entries with "content_type": "Note" is correct, is not 14, but 31. So maybe the offline decrypt tool can be updated to decode correctly the files saved after Jan 8, 2021.

**Comment by mobitar** [comment]
Thanks for the report. Can replicate this, investigating..

**Comment by mobitar** [comment]
@Matei59 have found and fixed the issue. Will likely deploy the fix sometime this week. It's an issue with client-side processing which affects the applications and offline decrypt script, and not an issue with the server-side backups.

Until then if you'd like a backup that is sure to work for your peace of mind, you can download an encrypted backup using the applications directly 👍

**Comment by Matei59** [comment]
Thank you for the quick and helpful feedback. I am looking forward to the fix. 
Do I understand correctly that after the issue is resolved, the Dropbox backups from Jan 8 to present will be correctly and fully decoded? (using a new version of the offline decrypt script) 
Could you please confirm?
Many thanks again.

**Comment by mobitar** [comment]
Correct, after we release a new version of the decrypt script + applications, you will be able to use them to correctly decrypt your existing Dropbox backups.

In addition, backups made today using the current applications will work with the present decrypt script with no problem.

**Comment by mobitar** [comment]
This issue has been fixed in the [latest decrypt script](https://github.com/standardnotes/decrypt) and also desktop/web application version 3.6.13.
